src/sidebar/components/MarkdownView.tsx

```diff
@@ -35,21 +76,49 @@ export default function MarkdownView({
     });
   }, [markdown]);
 
+  useEffect(() => {
+    const listenerCollection = new ListenerCollection();
+    const foundMentions = replaceMentionTags(content.current!, mentions);
```

Comment:
Similarly to what we do to replace links with embeds via a side effect, we replace mention tags with their rendered representation directly in the DOM.